From ecef45e5801e8a41508ad4937784ba19f545eb88 Mon Sep 17 00:00:00 2001
From: George G <kreach3r@users.noreply.github.com>
Date: Mon, 14 Nov 2016 14:49:47 +0200
Subject: [PATCH 18/24] Themes: Notification dynamic colors bool compatible
 with OMS7

forward ported to oreo

OMS7 introduced this fine piece of code: https://github.com/SubstratumResources/platform_frameworks_base/blob/n-oms7/core/java/android/app/ResourcesManager.java#L897..#L904

// Resources.getSystem Resources are created on request and aren't tracked by
// mResourceReferences.
//
// If overlays targeting "android" are to be used, we must create the system
// resources regardless of whether they already exist, since otherwise the
// information on what overlays to use would be lost. This is wasteful for most
// applications, so limit this operation to the system user only. (This means
// Resources.getSystem() will *not* use overlays for applications.)

Replaced deprecated Resources.getSystem() with compatible method.

Change-Id: I02efe27de3cc7067552964ffbaf079f9e9b5bc3e
---
 core/java/android/app/Notification.java | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/core/java/android/app/Notification.java b/core/java/android/app/Notification.java
index 7489d95..4f9d804 100644
--- a/core/java/android/app/Notification.java
+++ b/core/java/android/app/Notification.java
@@ -4735,7 +4735,7 @@ public class Notification implements Parcelable
                 boolean ambient) {
             boolean colorable = !isLegacy() || getColorUtil().isGrayscaleIcon(mContext, smallIcon);
             int color;
-            if (!Resources.getSystem().getBoolean(R.bool.config_allowNotificationIconTextTinting)) {
+            if (!mContext.getResources().getBoolean(R.bool.config_allowNotificationIconTextTinting)) {
                 color = ambient ? resolveAmbientColor() : mContext.getColor(R.color.notification_icon_default_color);
             } else {
                 color = ambient ? resolveAmbientColor() : getPrimaryHighlightColor();
@@ -4774,7 +4774,7 @@ public class Notification implements Parcelable
         }
 
         int resolveIconContrastColor() {
-            if (!Resources.getSystem().getBoolean(R.bool.config_allowNotificationIconTextTinting)) {
+            if (!mContext.getResources().getBoolean(R.bool.config_allowNotificationIconTextTinting)) {
                 return mContext.getColor(R.color.notification_icon_default_color);
             } else {
                 return resolveContrastColor();
@@ -4782,7 +4782,7 @@ public class Notification implements Parcelable
         }
 
         int resolveContrastColor() {
-            if (!Resources.getSystem().getBoolean(R.bool.config_allowNotificationIconTextTinting)) {
+            if (!mContext.getResources().getBoolean(R.bool.config_allowNotificationIconTextTinting)) {
                 return mContext.getColor(R.color.notification_text_default_color);
             }
 
-- 
2.1.2

